workflows/templates/solver-template.yaml

```diff
@@ -0,0 +1,80 @@
+---
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: solver-any
+  annotations:
+    thoth-station.ninja/template-version: 0.10.0-dev
+  labels:
+    app: thoth
+    component: solver
+spec:
+  templates:
+    - name: solve
+      inputs:
+        parameters:
+          - name: "THOTH_SOLVER_NAME"
+          - name: "THOTH_SOLVER_DOCUMENT_ID"
+          - name: "THOTH_LOG_SOLVER"
+            value: "INFO"
+          - name: "THOTH_SOLVER_NO_TRANSITIVE"
+            value: "1"
+          - name: "THOTH_SOLVER_PACKAGES"
+          - name: "THOTH_SOLVER_INDEXES"
+      outputs:
+        artifacts:
+          - name: solver-document
+            path: "/mnt/workdir/{{inputs.parameters.THOTH_SOLVER_DOCUMENT_ID}}"
+            archive:
+              none: {}
+            s3:
+              key: "argo/solver/{{inputs.parameters.THOTH_SOLVER_DOCUMENT_ID}}"
+              endpoint: "s3.upshift.redhat.com"
+              bucket: "thoth"
+              insecure: true
+              accessKeySecret:
+                name: argo-artifact-repository
+                key: accessKey
+              secretKeySecret:
+                name: argo-artifact-repository
+                key: secretKey
+      container:
+        name: solver
+        image: "solver-{{inputs.parameters.THOTH_SOLVER_NAME}}"
+        env:
+          - name: THOTH_SOLVER
+            value: "solver-{{inputs.parameters.THOTH_SOLVER_NAME}}"
```

Comment:
Will this become something like `solver-solver-fedora-31-py37`?

Reply:
Good point, depends on the input. I tested using `fedora-31-py37` as input, so if the input is with solver in front, then the template need to be updated :)